lib/battle/CBattleInfoCallback.cpp

```diff
@@ -1448,11 +1435,10 @@ AttackableTiles CBattleInfoCallback::getPotentiallyShootableHexes(const battle::
 	AttackableTiles at;
 	RETURN_IF_NOT_BATTLE(at);
 
-	if(attacker->hasBonusOfType(BonusType::SHOOTS_ALL_ADJACENT) && !vstd::contains(attackerPos.neighbouringTiles(), destinationTile))
+	if(attacker->hasBonusOfType(BonusType::SHOOTS_ALL_ADJACENT) && !BattleHexArray::neighbouringTilesCache[attackerPos].contains(destinationTile))
```

Comment:
Perhaps attackerPos.neighbouringTiles() should return `BattleHexArray::neighbouringTilesCache[pos]` instead? 

Old version of this code is more clear - user code should not be aware of implementation details.